kazoo/tests/test_watchers.py

```diff
@@ -171,6 +171,46 @@ def changed(d, stat):
 
         eq_(args, [None, None])
 
+    def test_no_such_node_for_children_watch(self):
+        args = []
+        path = self.path + '/test_no_such_node_for_children_watch'
+        update = threading.Event()
+
+        def changed(children):
+            args.append(children)
+            update.set()
+
+        # watch a node which does not exist
+        children_watch = self.client.ChildrenWatch(path, changed)
+        eq_(update.is_set(), False)
+        eq_(children_watch._stopped, True)
+        eq_(args, [])
+
+        # watch a node which exists
+        self.client.create(path, b'')
+        children_watch = self.client.ChildrenWatch(path, changed)
+        update.wait(3)
+        eq_(args, [[]])
+        update.clear()
+
+        # watch changes
+        self.client.create(path + '/fred', b'')
+        update.wait(3)
+        eq_(args, [[], ['fred']])
+        update.clear()
+
+        # delete children
+        self.client.delete(path + '/fred')
+        update.wait(3)
+        eq_(args, [[], ['fred'], []])
+        update.clear()
+
+        # delete watching
+        self.client.delete(path)
+        time.sleep(1)
```

Comment:
It'd be nice if there was a non-timing dependent way to wait for this. But I understand there may not be. Maybe if the test tries to grab the watchers _run_lock which should be released as soon as _stopped is set and returned?